### DISCUSSION TITLE
lath martensite microstructure

### DISCUSSION THREAD

**Discussion by fischertim1** [mtex-toolbox/mtex#1161] [Ask Anything] [answered]
Hello,

I am trying to create a synthetic lath martensite microstructure. Therefore, I'm using Neper is a software package for polycrystal generation.
Is there a way in MTEX to compute crystal orientations of the individual laths with the Kurdjumov-Sachs orientation relationship?

Many thanks,
Tim

**Comment by nyyssont** [comment, marked as answer]
Hello Tim,

it is possible. An example is below and should produce the figure shown.

<details><summary><b>Click to view script</b></summary>
<p>

```matlab
cs_bcc = crystalSymmetry('m-3m', [2.9 2.9 2.9],...
    'mineral', 'Iron bcc (old)', 'color', [0.53 0.81 0.98]);
cs_fcc = crystalSymmetry( 'm-3m', [3.7 3.7 3.7],...
    'mineral', 'Iron fcc', 'color', [0.56 0.74 0.56]);

ori_aus = orientation.rand(1,cs_fcc);

p2c = orientation.KurdjumovSachs(cs_bcc,cs_fcc);

ori_mar = variants(p2c, ori_aus);

h_bcc = Miller({1,0,0},{1,1,0},{1,1,1},cs_bcc);
h_fcc = Miller({1,0,0},{1,1,0},{1,1,1},cs_fcc);

plotPDF(ori_mar,h_bcc,'MarkerSize',5,'MarkerFaceAlpha',0.05,'MarkerEdgeAlpha',0.1);

nextAxis(1)
hold on
plot(ori_aus * h_fcc(1).symmetrise ,'MarkerFaceColor','r')
xlabel('\((100)\)','Color','red','Interpreter','latex')

nextAxis(2)
plot(ori_aus * h_fcc(3).symmetrise ,'MarkerFaceColor','r')
xlabel('\((111)\)','Color','red','Interpreter','latex')

nextAxis(3)
plot(ori_aus * h_fcc(2).symmetrise ,'MarkerFaceColor','r')
xlabel('\((110)\)','Color','red','Interpreter','latex')
hold off
```

</p>
</details>

![example](https://user-images.githubusercontent.com/10060198/127640906-6de11777-922a-4e4c-ad3a-1295d4cf1c65.jpg)

Please see the following for additional explanations:
https://mtex-toolbox.github.io/ParentChildVariants.html
https://mtex-toolbox.github.io/MartensiteVariants.html

This should help you get started.

**Reply by fischertim1** [reply]
Hej Tuomo,

thank you for the script and the quick reply! That is very useful. 
Is there also a way, based on the given PAG orientation, to calculate the lattice normals of the four possible variants for each packet. 
Hope that question makes sense. I'm not very experienced with MTEX.

Best regards,
Tim

**Reply by nyyssont** [reply]
I think it might be possible. However, could you describe a bit more in detail what sort of data you would be expecting. By lattice normals, do you mean the normals of the parallel lattice planes of child and parent phases? How would you like them to be described (vectors in specimen coordinate system, Euler angles, or what?)

**Reply by fischertim1** [reply]
Sure, please excuse my vague description! 
For the artificial generation of lath martensitic microstructure, I need the normal direction of the four possible habit planes in vector format. For each packet, I want to randomly select one of the four habitat planes. The individual blocks within a packet should then be generated in this normal direction.
I hope this makes more sense now.

**Reply by nyyssont** [reply]
Let's make a rough assumption that the habit planes are parallel to the {111} planes in austenite. Then the vectors making up the red dots in the center pole figure above are the habit plane normals. 

The code below will show how to get the habit plane normals as vectors `h_pl` and how they relate to each variant.

<details><summary><b>Click to view script</b></summary>
<p>

```matlab
cs_bcc = crystalSymmetry('m-3m', [2.9 2.9 2.9],...
    'mineral', 'Iron bcc (old)', 'color', [0.53 0.81 0.98]);
cs_fcc = crystalSymmetry( 'm-3m', [3.7 3.7 3.7],...
    'mineral', 'Iron fcc', 'color', [0.56 0.74 0.56]);

ori_aus = orientation.rand(1,cs_fcc);

p2c = orientation.KurdjumovSachs(cs_fcc,cs_bcc);

ori_mar = variants(p2c, ori_aus);

h_bcc = Miller({1,0,0},{1,1,0},{1,1,1},cs_bcc);
h_fcc = Miller({1,0,0},{1,1,0},{1,1,1},cs_fcc);

%Let's make a rough assumption that the habit planes are the 111 planes of
%austenite:
h = Miller({1,1,1},{1,-1,1},{-1,1,1},{1,1,-1},p2c.CS);

%habit planes in specimen coordinate system:
h_pl = ori_aus.project2FundamentalRegion.*h

hold on
plot(h_pl,'plane')
plot(h_pl,'x')

figure
plotPDF(ori_mar(1:6),h_bcc(1),...
    'DisplayName','packet 1, h_pl(1)');
hold on
plotPDF(ori_mar(7:12),h_bcc(1),...
    'DisplayName','packet 2, h_pl(2)');
plotPDF(ori_mar(13:18),h_bcc(1),...
    'DisplayName','packet 3, h_pl(3)');
plotPDF(ori_mar(19:24),h_bcc(1),...
    'DisplayName','packet 4, h_pl(4)');
legend
```

</p>
</details>

We generated 24 variant orientations for martensite. The variants are ordered according to the Morito conventions, which means that the first six variants would form on `h_pl(1)`, the next six variants on `h_pl(2)`, the next six on `h_pl(3)` and the last six on `h_pl(4)`. The following figure shows the variants colored according to habit plane (as defined above).

![habit_plane](https://user-images.githubusercontent.com/10060198/153464962-308afbcd-b871-43b3-bc4a-38ce722ce7d5.jpg)

**Reply by fischertim1** [reply]
Excellent! That helps me a lot. I appreciate your help.

**Reply by danishTUE** [reply]
Hello @nyyssont,

I am trying to learn MTex and how its used to calculate parent to child orientational relationship. I came across this discussion and had a query about it. In the example script you gave in your first response, shouldnt the command to get the KS misorientation be  "p2c = orientation.KurdjumovSachs(**_cs_fcc,cs_bcc_**);" (as described here: [https://mtex-toolbox.github.io/MartensiteVariants.html](url) ) instead of "p2c = orientation.KurdjumovSachs(**_cs_bcc,cs_fcc_**);"? Or, am I wrong? Kindly clarify.

Many thanks in advance.

**Reply by nyyssont** [reply]
Yep, you are right. It was a mistake. Although, it seems the orientations produced by the command are the same, even if the crystal symmetries are mixed up (order is changed).

Generally speaking, I would rely on the commands in the help files. Just to be clear, `p2c = orientation.KurdjumovSachs(cs_fcc,cs_bcc)` is the correct syntax.
